File: 3838-weighted-word-mapping/solution.py
class Solution:
    def weightedWordMapping(self, words: list[str], weights: list[int]) -> str:
        result = []
        
        for word in words:
            # Calculate total weight of the word
            total_weight = sum(weights[ord(char) - ord('a')] for char in word)
            
            # Map the weight modulo 26 to a char in reverse alphabetical order
            # 0 -> 'z', 1 -> 'y', ..., 25 -> 'a'
            remainder = total_weight % 26
            mapped_char = chr(ord('z') - remainder)
            
            result.append(mapped_char)
            
        return "".join(result)

File: 3838-weighted-word-mapping/solution.cpp
#include <vector>
#include <string>

class Solution {
public:
    std::string weightedWordMapping(std::vector<std::string>& words, std::vector<int>& weights) {
        std::string result = "";
        
        for (const std::string& word : words) {
            int totalWeight = 0;
            // Calculate total weight of the word
            for (char c : word) {
                totalWeight += weights[c - 'a'];
            }
            
            // Map the weight modulo 26 to a char in reverse alphabetical order
            // 0 -> 'z', 1 -> 'y', ..., 25 -> 'a'
            int remainder = totalWeight % 26;
            char mappedChar = 'z' - remainder;
            
            result += mappedChar;
        }
        
        return result;
    }
};